### DISCUSSION TITLE
Extension / Plugin support for additional use cases?

### DISCUSSION THREAD

**Discussion by julbrs** [renovatebot/renovate#22715] [Suggest an Idea]
### Tell us more.

We are using Renovate w/ the self-hosting option on our Bitbucket cloud repo with a great success.
 Around 30 repos are scanned and PR are create automatically, some teams have already enable auto-merge!

I have an additional requirement, I want to list dependencies for all my repos. Ideally, I would like to be able to put some data on each dep: why are we using this dependency, what is coupling level? This little tool help me: https://s-maciejewski.github.io/pretty-package/

I was wondering if Renovate can be 'extended' in some way as it already 'know' the list of dependencies for all my projects ;) Then it can allow fellows like me to build a plugin that can fit a very specific purpose (document all the dependencies here).

Thanks!

**Comment by viceice** [comment]
you find that info inside debug logs, see package files with updates message.

you can also log to file where each line is a json object for simpler parsing , aka ndjson

**Comment by Churro** [comment]
Could this project by @jamietanna probably be what you're looking for? https://dmd.tanna.dev

**Reply by jamietanna** [reply]
Thanks @churro! Yes absolutely this could be used - @julbrs let me know if you'd like to chat about getting it set up, it uses Renovate under the hood ([more details](https://github.com/renovatebot/renovate/discussions/13150#discussioncomment-4029477)) and produces you an SQLite database for further querying

**Reply by julbrs** [reply]
thanks @Churro & @jamietanna!

That's a really good track to follow! I will spend some time to understand how @jamietanna have built this project ; and see how I will handle that. I have a few more requirements (adding some comments on every single dependency) ; but maybe I can work around https://dmd.tanna.dev/ and make some improvements?

Give me some time to dig that first 🤓

**Reply by jamietanna** [reply]
Nice! Happy to move to a different discussion channel, i.e. over email, if that'd be helpful. But very positive about getting more usage out of the project, and that it will hopefully give you what you want, or we can look to add it :)
